engines/gob/expression.cpp

```diff
@@ -231,6 +231,41 @@ void Expression::printExpr_internal(char stopToken) {
 	while (true) {
 		operation = _vm->_game->_script->readByte();
 
+		while ((operation == 14) || (operation == 15)) {
+			if (operation == 14) {
+				// Add a direct offset
+				debugN(5, "#%d#", _vm->_game->_script->readUint16());
+				_vm->_game->_script->skip(2);
+
+				if (_vm->_game->_script->peekByte() == 97)
+					_vm->_game->_script->skip(1);
+			} else if (operation == 15) {
+				// Add an offset from an array
+				debugN(5, "#%d", _vm->_game->_script->readUint16());
+				_vm->_game->_script->skip(2);
+
+				dimCount = _vm->_game->_script->readByte();
+				for (int i = 0; i < dimCount; i++)
+					debugN(5, "[]");
+				_vm->_game->_script->skip(dimCount);
+
+				debugN(5, "->");
+
+				for (int i = 0; i < dimCount; i++) {
+					debugN(5, "{");
+					printExpr_internal(OP_END_MARKER);
+					debugN(5, "->");
+				}
+
+				debugN(5, "#");
+
+				if (_vm->_game->_script->peekByte() == 97)
+					_vm->_game->_script->skip(1);
+			}
+
+			operation = _vm->_game->_script->readByte();
+		}
```

Comment:
```suggestion
		}
		
		debugN(5, "\n");
```

So, you have your debug array print out completed

Reply:
I do not think it makes sense to add a line break here - it would cut the expression before adding an offset value after the '#'. The parsed expression seems to be properly displayed in the debug output in all cases.